README.md

```diff
@@ -1,2 +1,119 @@
-# spire-mysql-demo
-Demonstration of using SPIRE-issued X.509-SVIDs to set up an authenticated mTLS connection to a MySQL database
+# MySQL mTLS Authentication using SPIRE
+
+Demonstration of using SPIRE-issued X.509-SVIDs in MySQL database and clients, providing client authentication
+and connection encryption in MySQL via mutual-TLS (mTLS), deployed on Kubernetes. This project assumes readers to have 
+basic knowledge of the following technologies and concepts:
+
+1. [SPIFFE](https://spiffe.io/) and [SPIRE](https://github.com/spiffe/spire)
+2. [Kubernetes](https://kubernetes.io/)
+3. [MySQL](https://www.mysql.com/)
+4. [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security)
+
+## [Getting Started](#getting-started)
+
+This section guides deploying and setting up components involved in this project on a Kubernetes cluster.
+Refer [design](#design) for details on the components involved.
+
+### Prerequisites
+
+1. `kubectl` is installed and available on the PATH: https://kubernetes.io/docs/tasks/tools/ 
+2. Kubernetes cluster is configured with `kubectl` and `kubectl context` is set to use this cluster
+3. `curl` is installed and available on the PATH: https://everything.curl.dev/get
```

Comment:
I think curl.se is the official website for cURL, referred to in the source code repo on GitHub: https://github.com/curl/curl#website

Maybe we should link to the official site?